### PR TITLE
Ros2 migration behavior/arm

### PR DESCRIPTION
**Pull request by ana-GT**
Conversion was straightforward, with the exception of a few issues that I marked TODO:

1. arm/tools/arm_tool.cc, line 168: The arm_tool is initialized using the AnonymousName option, which is not yet available in ROS2, although there seems to be active work on it (as of this week).
2. arm/src/arm_nodelet.cc, lines 491-493: Some functions were not ported, as they no longer seem applicable in ROS2, such as GetPrivateHandle(), Listen function for reconfiguring parameters and its corresponding reconfigure_callback function.
3. arm/src/arm_nodelet.cc: Line 546: A subscriber for TOPIC_JOINT_GOALS is created, originally with latch=true, however  this topic is not in the LatchedTopic list, so if we use the FF_CREATE_SUBSCRIBER, the topic will not be latched.

### CODE REVIEW

**Comment by trey0** [comment]
> 1. arm/tools/arm_tool.cc, line 168: The arm_tool is initialized using the AnonymousName option, which is not yet available in ROS2, although there seems to be active work on it (as of this week).

As far as I understand, the `AnonymousName` option appends a random number to the end of the node name at initialization time, which we could do manually at [1]. I see you also changed the node name from `"control"` to `"arm_tool"`, which seems like an improvement.

[1] https://github.com/nasa/astrobee/blob/939adfe9cb50e536800afb6937c21793a64e3e41/behaviors/arm/tools/arm_tool.cc#L195

**Comment by trey0** [comment]
> 3. arm/src/arm_nodelet.cc: Line 546: A subscriber for TOPIC_JOINT_GOALS is created, originally with latch=true, however  this topic is not in the LatchedTopic list, so if we use the FF_CREATE_SUBSCRIBER, the topic will not be latched.

@marinagmoreira and @kbrowne15 recently merged https://github.com/nasa/astrobee/commit/a3ae21283d311fcc9c48d3cf75fb4d76bc41ed07 on `develop`, which makes this publisher non-latching.

I'm not sure why it was originally marked latching. The latching option would have the effect of automatically resending the joint goals if somehow the arm nodelet joined late after the goals were originally published, but I don't think that's really desired (could lead to an unintended "zombie command").

**Comment by marinagmoreira** [comment]
So I tested the arm behavior in simulation and it works. But let's hold on merging this until we make a decision on the services since it affects it.
I commented out the services test because as a gtest, it's really hard to test it since the server and client are started from the same main, and that opens a can of worms for the sync service problem. That's why the service test passed in the past, even if we couldn't use it for components.
I'll keep on tinkering with the service test to see if I can make it work with a multithreaded executor since we might want to have the server and client under the same container and if this doesn't work, that's just problems down the line.. just don't want to sink even more time into it :(..

File: behaviors/arm/src/arm_nodelet.cc [hunk 1]
```diff
@@ -507,83 +536,78 @@ class ArmNodelet : public ff_util::FreeFlyerNodelet {
...
+        std::bind(&ArmComponent::TimeoutCallback, this), nh, true, false);
 
     // Publishers for arm and joint state
-    sub_joint_states_ = nh->subscribe(TOPIC_JOINT_STATES, 1,
-      &ArmNodelet::JointStateCallback, this);
-    pub_joint_goals_ = nh->advertise<sensor_msgs::JointState>(
-      TOPIC_JOINT_GOALS, 1);
+    sub_joint_states_ = FF_CREATE_SUBSCRIBER(nh, sensor_msgs::JointState, TOPIC_JOINT_STATES, 1,
+      std::bind(&ArmComponent::JointStateCallback, this, std::placeholders::_1));
+    // TODO(ana): This had a latch=true argument, but TOPIC_JOINT_GOALS is NOT in the LatchedTopic list
```
Comment (marinagmoreira) [inline behaviors/arm/src/arm_nodelet.cc:549]: According to the analysis in https://github.com/nasa/astrobee/issues/628, this does not need to be latched

File: behaviors/arm/src/arm_component.cc [hunk 2]
```diff
@@ -17,43 +17,66 @@
...
-  };
+enum ServoID {
+  PROXIMAL_SERVO,           // Proximal joint servo
+  DISTAL_SERVO,             // Distal joint servo
+  GRIPPER_SERVO,            // Gripper joint servo
+  ALL_SERVOS          // Proximal, Distal and Gripper servos
+};
 
 // Joint information, where HUMAN = SCALE * DRIVER + OFFSET
 struct JointInfo {
```
Comment (bcoltin) [inline behaviors/arm/src/arm_component.cc:82]: Now that we aren't using tol (I'm pretty sure) except to set it but never look at it, I would just delete it from JointInfo to make things more clear.

**Review by bcoltin — APPROVED** [review]
Thanks Ana and Marina!

File: behaviors/arm/src/arm_component.cc (hunk 2 above)
Comment (trey0) [reply behaviors/arm/src/arm_component.cc:82]: That sounds like a change to apply/test on `develop` first?

Comment (bcoltin) [reply behaviors/arm/src/arm_component.cc:82]: I thought it only applies here due to changes in how the parameter server works, but correct me if I'm wrong Marina.

Comment (marinagmoreira) [reply behaviors/arm/src/arm_component.cc:82]: In order to maintain functionality with the previous version the tol with GetTolerance() still only gets updated in certain flight modes, which maintains the need for the tol variable :)

**Pull request merged**